### PR TITLE
Add `ABORTED` threshold to ReverseBuildTrigger

### PR DESCRIPTION
**Pull request by madsjakobsen**
Adds `ABORTED` as additional threshold to ReverseBuildTrigger to allow for triggering builds even if the watched project is aborted. 

### Proposed changelog entries

* Add `ABORTED` threshold to `ReverseBuildTrigger`.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Comment by daniel-beck** [comment]
Could you provide more information about when these would be needed? It seems like a _very_ niche use case for one, and I cannot think of a use case for the other.

**Comment by MadsJakobsen** [comment]
> Could you provide more information about when these would be needed? It seems like a _very_ niche use case for one, and I cannot think of a use case for the other.

Sorry I should have included the use case, at work we use jobs which triggers on the completion of other to perform certain tasks such as distribution of the artifacts, and notification. We do this to avoid touching the upstream pipelines, however the downstream jobs are currently not triggered when developers abort the jobs. 

I added `NOT_BUILT` for completion, I can remove that if's just noise

**Review by oleg-nenashev — APPROVED** [review]
Sorry for missing this PR. The change looks good to me. It might help users in some cases as @MadsJakobsen mentioned, and two extra states don not create issues for a config item hidden by default. Ideally we should rework the configurator to dropdown later

**Comment by daniel-beck** [comment]
@oleg-nenashev Could you clarify how `NOT_BUILT` could trigger?

**Comment by oleg-nenashev** [comment]
It couldn't. The same time I see no problem with it being added for consistency purposes

**Review by daniel-beck — CHANGES_REQUESTED** [review]
Since Oleg confirmed that this cannot happen: We should not add an option that is never useful or relevant.

**Comment by MadsJakobsen** [comment]
> Since Oleg confirmed that this cannot happen: We should not add an option that is never useful or relevant.

I was unaware that it couldn't happen, I absolutely agree, I will remove it 👍

**Comment by jglick** [comment]
> jobs which triggers on the completion of other to perform certain tasks such as distribution of the artifacts, and notification

Do I understand correctly that for purposes of `ABORTED`, only the latter use case (notification) makes sense? And for that use case you would potentially want to trigger on _all_ exit statuses including `NOT_BUILT` (most commonly encountered when using the `milestone` step).

**Comment by MadsJakobsen** [comment]
> > jobs which triggers on the completion of other to perform certain tasks such as distribution of the artifacts, and notification
> 
> Do I understand correctly that for purposes of `ABORTED`, only the latter use case (notification) makes sense? And for that use case you would potentially want to trigger on _all_ exit statuses including `NOT_BUILT` (most commonly encountered when using the `milestone` step).

Yes thats correct, thats a much better way of explaining it, I would like it to trigger on all exit statuses for the purpose of notifications. And I can see that `ABORTED` on it's own (as it is in the change right now) would be misleading since it would include `NOT_BUILT`.

File: core/src/main/resources/jenkins/triggers/ReverseBuildTrigger/config.jelly
Comment (jglick) [inline core/src/main/resources/jenkins/triggers/ReverseBuildTrigger/config.jelly:45]:
```suggestion
               title="${%Always trigger, even if the build is aborted}" value="ABORTED"/>
```

?

Comment (MadsJakobsen) [reply core/src/main/resources/jenkins/triggers/ReverseBuildTrigger/config.jelly]: I like that, I will commit the suggested 👍

**Review by jglick — APPROVED** [review]
AFAICT this mode was always an option when using Pipeline `properties` to set up the trigger, `job-dsl` plugin, etc.

**Review by daniel-beck — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Review by oleg-nenashev — APPROVED** [review]

**Pull request merged**
